ls1intum/Artemis
# `Development`: Refactor programming exercise grading service

### Checklist
#### General


#### Server

### Motivation and Context


The score calculation for programming exercises has become too complicated for both students,
developers and instructors.
This is a first step to understand the logic we currently have so that we can decide how to further simplify these calculations.

### Description
Multiple minor refactorings

### Steps for Testing
No new features -> Submit a programming exercise and confirm that the point calculation works as before.

Make sure that specific features are tested in various combinations
- Submission policies
- Static code analysis feedback
- Manual feedback
- Structured grading instructions

### Exam Mode Testing
Cover the same feature combinations as above however in the exam mode both as test run and actual exam

### Review Progress






#### Code Review
- [x] @Strohgelaender 
- [x] @b-fein 
- [ ] @JohannesStoehr 

#### Testing
- [x] @kekskatze 
- [x] @frederik-kt 
- [ ] @1999Markus 

#### Exam mode test
- [x] @kekskatze 
- [x] @simon-hng 

### Test Coverage
Refactoring only, no test changes.
